src/studio/src/designer/backend/Repository/ReleaseRepository.cs

```diff
@@ -1,57 +1,220 @@
 using Altinn.Studio.Designer.TypedHttpClients.AzureDevOps.Enums;
-using Microsoft.Azure.Documents;
+using Altinn.Studio.Designer.ViewModels.Request;
+using Altinn.Studio.Designer.ViewModels.Request.Enums;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Npgsql;
+using NpgsqlTypes;
 
 namespace Altinn.Studio.Designer.Repository
 {
     /// <summary>
-    /// ReleaseRepository
+    /// Handles release repository. 
     /// </summary>
-    public class ReleaseRepository : DocumentRepository
+    [ExcludeFromCodeCoverage]
+    public class ReleaseRepository : IReleaseRepository
     {
+        private readonly string insertReleaseSql = "call designer.insert_release(@id, @tagName, @org, @app, @buildId, @buildStatus, @buildResult, @created, @entity)";
+        private readonly string getReleasesSql = "select designer.get_releases(@_org, @_app, @_limit, @_order_asc_desc)";
+        private readonly string checkExistingReleaseSql = "select designer.check_existing_release(@_org, @_app, @_tagName, @_buildStatus, @_buildResult)";
+        private readonly string getReleaseSql = "select designer.get_release(@_org, @_buildId)";
+        private readonly string updateReleaseBuildSql = "call designer.update_release_build(@_id, @_buildStatus, @_buildResult, @_entity)";
+        private readonly string _connectionString;
+        private readonly ILogger _logger;
+
         /// <summary>
-        /// Constructor
+        /// Initializes a new instance of the <see cref="ReleaseRepository"/> class.
         /// </summary>
-        public ReleaseRepository(
-            IOptions<AzureCosmosDbSettings> options,
-            IDocumentClient documentClient)
-            : base(options.Value.ReleaseCollection, options, documentClient)
+        public ReleaseRepository(IOptions<PostgreSQLSettings> postgresSettings, ILogger<ReleaseRepository> logger)
         {
+            _connectionString = string.Format(
+                postgresSettings.Value.ConnectionString,
+                postgresSettings.Value.DesignerDbPwd);
+            _logger = logger;
         }
 
-        /// <summary>
-        /// Gets the succeeded release
-        /// </summary>
-        /// <param name="org">Organisation</param>
-        /// <param name="app">App</param>
-        /// <param name="tagName">Tag name</param>
-        /// <returns></returns>
-        public async Task<ReleaseEntity> GetSucceededReleaseFromDb(string org, string app, string tagName)
+        /// <inheritdoc/>
+        public async Task<ReleaseEntity> Create(ReleaseEntity releaseEntity)
         {
-            SqlQuerySpec sqlQuerySpec = CreateSqlQueryToGetSucceededRelease(org, app, tagName);
-            IEnumerable<ReleaseEntity> releases = await GetWithSqlAsync<ReleaseEntity>(sqlQuerySpec, org);
-            return releases.Single();
+            try
+            {
+                _logger.LogDebug("ReleasEntity: " + JsonString(releaseEntity));
```

Comment:
do we need this debug line now?

Reply:
It doesn't do any harm since it only logs if debug logging is enabled.